Buildings/Fluid/DXSystems/Heating/BaseClasses/CoilDefrostTimeCalculations.mo

```diff
@@ -0,0 +1,145 @@
+within Buildings.Fluid.DXSystems.Heating.BaseClasses;
+block CoilDefrostTimeCalculations
+  "Calculates defrost curve value at given temperature and mass flow rate"
+  extends Modelica.Blocks.Icons.Block;
+
+  parameter Buildings.Fluid.DXSystems.Heating.BaseClasses.Types.DefrostTimeMethods
+    defTri=Buildings.Fluid.DXSystems.Heating.BaseClasses.Types.DefrostTimeMethods.timed
+    "Type of method to trigger the defrost cycle";
+
+  parameter Real tDefRun(
+    final unit="1") = 0.5
+    "Time period fraction for which defrost cycle is run"
+    annotation(Dialog(enable=defTri == Buildings.Fluid.DXSystems.Heating.BaseClasses.Types.DefrostTimeMethods.timed));
+
+  parameter Modelica.Units.SI.ThermodynamicTemperature TDefLim
+    "Maximum temperature at which defrost operation is activated";
+
+  parameter Modelica.Units.SI.TemperatureDifference dTHys = 0.5
+    "Temperature comparison hysteresis difference"
+    annotation(Dialog(tab="Advanced"));
+
+  Modelica.Blocks.Interfaces.RealInput TOut(
+    final unit="K",
+    displayUnit="degC",
+    final quantity="ThermodynamicTemperature")
+    "Temperature of outdoor air"
+    annotation (Placement(transformation(extent={{-120,10},{-100,30}}),
+      iconTransformation(extent={{-120,10},{-100,30}})));
+
+  Modelica.Blocks.Interfaces.RealInput XOut(
+    final unit="kg/kg",
+    final quantity="MassFraction")
+    "Humidity ratio of outdoor air"
+    annotation (Placement(transformation(extent={{-120,-30},{-100,-10}}),
+      iconTransformation(extent={{-120,-30},{-100,-10}})));
+
+  Modelica.Blocks.Interfaces.RealOutput tDefFra(
+    final unit="1")
+    "Defrost time period fraction"
```

Comment:
time period fraction is not clear. Explain what it is a fraction off, what is the basis, e.g., what would a value of 1 mean?